hedwig.py

```diff
@@ -0,0 +1,344 @@
+    choices = {}
+    for key, val in current_list.items():
+        if label.lower() == val.lower():
+            choices[key] = val
+
+    if len(choices) == 1:
+        for key in choices:
+            return key
+    else:
+        #TODO: deal with duplicates
+        return None
+
+def check_filter(label, category):
+    cleanfig_path = os.path.join("filters", "extras.yaml")
+    with open(cleanfig_path, 'r') as config_file:
+        cleanfig = yaml.load(config_file.read())
+    #primary cleaning
+    abbrev_table = cleanfig.get('abbrev_table')
+    label += " " #add trailing space in case abbreviation is end of string
+    label = label.replace("  ", " ")
+    label = label.replace("\\", "") #can not have escape character when using update api
+    for abbrev in abbrev_table:
+        label = label.replace(abbrev, abbrev_table[abbrev])
+    if label[-1] == " ":
+        label = label[:-1] #remove final space
+
+    #match against filter
+    filter_file = category + "_filter.txt"
+    filter_path = os.path.join("filters", filter_file)
+    pairs = {}
+    with open(filter_path) as fil:
+        content = fil.readlines()
+        for line in content:
+            key, val = line.split("|")
+            pairs[key] = val
+
+        for key, val in pairs.items():
+            if label == key:
+                label = val
+                break
+    
+    #return clean_label
+    return label
+
+def create_journal(connection, journal_name, publisher_name):
+    global PUBLISHER_COUNT
+    global JOURNAL_COUNT
+    clean_pname = check_filter(publisher_name.title(), 'publisher')
+    publisher_n = match_input(connection, clean_pname, 'publisher')
+
+    reqs = queries.make_journal.get_params(connection)
+    reqs['Journal'].name = journal_name
+    if publisher_n:
+        reqs['Publisher'].n_number = publisher_n
+    else:
+        reqs['Publisher'].name = clean_pname
+        PUBLISHER_COUNT += 1
+
+    response = queries.make_journal.run(connection, **reqs)
+    print(response)
+    if response:
+        JOURNAL_COUNT += 1
+        return reqs['Journal'].n_number
+    else:
+        return None
+
+def add_authors(connection, article, data):
+    global AUTHOR_COUNT
+    author_str = data['author']
+    authors = author_str.split(" and ")
+    for person in authors:
+        args = queries.add_author_to_pub.get_params(connection)
+        author_n = match_authors(connection, person, data)
+        args['Author'].n_number = author_n
+        args['Article'].n_number = article.n_number
+        args['Author'].name = person
+
+        old_author = False
+        if author_n:
+            old_author = queries.check_author_on_pub.run(connection, **args)
+        if not old_author:
+            response = queries.add_author_to_pub.run(connection, **args)
+            print(response)
+            if response:
+                AUTHOR_COUNT += 1 
+
+def match_authors(connection, label, data):
```

Comment:
too much is happening in this function. I am having a hard time keeping track of what is happening at a high level. Please split out into functions with descriptive names